game-core/src/main/java/games/strategy/triplea/ui/mapdata/IslandTerritoryFinder.java

```diff
@@ -0,0 +1,74 @@
+package games.strategy.triplea.ui.mapdata;
+
+import games.strategy.ui.Util;
+import java.awt.Polygon;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+class IslandTerritoryFinder {
+
+  /**
+   * Finds all island territories given a set of territory names and their corresponding polygons.
+   * An island is defined as a land territory that is contained within a sea territory.
+   *
+   * <p>Warning: territories may consist of multiple polygons, only the first polygon is checked.
+   *
+   * @param polygons Mapping of territory names to the polygons representing that territories.
+   * @return A mapping of sea territory names to the name of land territories that are islands
+   *     contained by each corresponding sea territory.
+   */
+  static Map<String, Set<String>> findIslands(final Map<String, List<Polygon>> polygons) {
```

Comment:
Another quick thought:
What about making this class non-static and make the polygon map an actual field?
This way a lot of lambdas could be turned into private methods and there would be no need to pass this map to almost every method.

Reply:
Not a bad thought, though I think that might be going the wrong direction. It's probably better to try and collapse methods where it makes sense rather than storing state.